### PR TITLE
add offhand support to placeBlockWithOptions

### PR DESCRIPTION
**Pull request by aesthetic0001**
Allows for ppl to use {offhand: true} as an option to place the block in the offhand slot when calling "placeBlockWithOptions"

### CODE REVIEW

File: lib/plugins/place_block.js [hunk 1]
```diff
@@ -7,7 +7,15 @@ function inject (bot, { version }) {
   const Item = require('prismarine-item')(version)
 
   async function placeBlockWithOptions (referenceBlock, faceVector, options) {
-    if (!bot.heldItem) throw new Error('must be holding an item to place a block')
+    // 0 is placing with mainhand whilst 1 is off-hand
```
Comment (u9g) [inline lib/plugins/place_block.js:10]: no need for the comments

**Review by u9g — CHANGES_REQUESTED** [review]
would be good to add to typings/api

File: lib/plugins/place_block.js (hunk 1 above)
Comment (Zn10plays) [reply lib/plugins/place_block.js:10]: Do we usually handle errors like this? In the contributors' guide it's says to return the error insted of throwing it.

**Comment by aesthetic0001** [comment]
> would be good to add to typings/api

I don't think there's docs for placeBlockWithOptions so should I add that along with the new option?

File: lib/plugins/place_block.js (hunk 1 above)
Comment (aesthetic0001) [reply lib/plugins/place_block.js:10]: @Zn10plays it does, but I just copied what was written to handle a scenario where bot.helditem is null, so should I change it?

**Review by KaffinPX — APPROVED** [review]
LGTM but adding types will helpful.

File: lib/plugins/place_block.js (hunk 1 above)
Comment (Zn10plays) [reply lib/plugins/place_block.js]: Up to you on this one because, .catch will work with either

**Comment by aesthetic0001** [comment]
Is there anything I should change for this to be approved?

**Comment by DrMoraschi** [comment]
Why no typings?

**Comment by aesthetic0001** [comment]
> Why no typings?

Check the contribute channel

**Comment by u9g** [comment]
I just started tests.

**Comment by aesthetic0001** [comment]
yay tests passed

**Comment by aesthetic0001** [comment]
why does it say "Merging can be performed automatically once the requested changes are addressed" even tho I made the changes a while back

**Comment by rom1504** [comment]
can you please run git pull and push again ?

**Comment by aesthetic0001** [comment]
yup should be done

**Comment by DrMoraschi** [comment]
Why the .idea folder?

**Review by rom1504 — CHANGES_REQUESTED** [review]
please remove .idea

**Comment by aesthetic0001** [comment]
bruh i forgot to rm it mb

**Review by rom1504 — CHANGES_REQUESTED** [review]
can you add the option in the doc ?

**Comment by aesthetic0001** [comment]
which file should it be added to bcs its a _ method and there aren't docs for it yet afaik

**Comment by rom1504** [comment]
true
I really don't like these non documented methods but not your fault

**Comment by u9g** [comment]
This can be added to the _ functions docs, its in docs/

**Comment by aesthetic0001** [comment]
> This can be added to the _ functions docs, its in docs/

https://github.com/PrismarineJS/mineflayer/blob/master/docs/unstable_api.md?

**Comment by rom1504** [comment]
let's do it in another pr

**Pull request merged**